docs/controllers/scaling.md

```diff
@@ -0,0 +1,73 @@
+# Scaling
+
+This chapter is about strategies for scaling controllers and the tradeoffs these strategies make.
+
+## Motivating Questions
+
+- Why is the reconciler lagging? Are there too many resources being reconciled?
+- What happens when your controller starts managing resource sets so large that it starts significantly impacting your cpu or memory use?
+- Do you give your more resources?
+- Do you add more pods? How can you do this safely?
+
+Scaling an efficient Rust application that spends most of its time waiting for network changes might not seem like a complicated affair, and indeed, you can scale a controller in many ways and achieve good outcomes. But in terms of costs, not all solutions are created equal; are you avoiding improving your algorithm, or are you throwing more expensive machines at the problem?
+
+## Scaling Strategies
+
+We recommend trying the following scaling strategies in order:
+
+1. [[#Controller Optimizations]] (minimize expensive work to allow more work)
+2. [[#Vertical Scaling]] (more headroom for the single pod)
+3. [[#Sharding]] (horizontal scaling)
+
+In other words, try to improve your algorithm first, and once you've reached a reasonable limit of what you can achieve with that approach, allocate more resources to the problem.
+
+### Controller Optimizations
+Ensure you look at common controller [[optimization]] to:
+
+* minimize network intensive operations
+* cache/memoize expensive work
+* checkpoint progress on `.status` objects to avoid repeating work
+
+When checkpointing, care should be taken to not accidentally break [[reconciler#idempotency]].
+
+### Vertical Scaling
+
+* increase cpu/memory limits
+* configure controller concurrency (as a multiple of CPU limits)
+
+The [controller::Config] by currently[**](https://github.com/kube-rs/kube/issues/1473) defaults to __unlimited concurrency__ and may need tuning for large workloads.
```

Comment:
nit: is the `by` in here a small typo? makes the text just a little bit harder to understand.

Reply:
almost certainly a buffer accidentally vomited in there, good spot